### ISSUE TITLE
Do not delete id from employee table

### ISSUE THREAD

**Issue by BenTen** [PrestaShop/PrestaShop#25973]
#### Is your feature request related to a problem?

When we delete an employee from the menu "Team" in the BO, the id is delete from the database _employee (ps 1.7.5)

#### Describe the solution you'd like

It will be better to add a column "deleted" and keep all id on the table _employee. Especially, when other table record the id of the employee (module_history for example)

#### Alternatives you've considered
N/A

#### Additional context
N/A

**Comment by matks** [comment]
So basically you're suggesting we implement a "soft delete" feature for employees?

**Comment by BenTen** [comment]
What do you mean?

**Comment by BenTen** [comment]
We have to deactivate the account but not delete on the database.

For example in module_history, we can see employee with id 5 installed the module "abc".
It's cool but if the employee table show only the id 1, who can know who is id 5?

**Comment by hibatallahAouadni** [comment]
Hello @BenTen 

> So basically you're suggesting we implement a "soft delete" feature for employees?

It means, in BO the admin can not see the employee but it's saved in the database.

Thank you for your suggestion. The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by matks** [comment]
> We have to deactivate the account but not delete on the database.

It's indeed a "soft delete" :D

**Comment by BenTen** [comment]
Understood, I didn't know this term :)
Seems a good resume for what I said :3

**Comment by boo-code** [comment]
Yes, a soft delete - and core already does exactly that for two other entities, so it is a pattern to extend rather than one to introduce.

State on `9.2.x`. `ps_employee` has an `active` column but no `deleted` one, and `Employee` does not override `delete()`, so deleting an employee goes through `ObjectModel::delete()` and removes the row outright.

The reporter's concern is measurable: **15 tables** carry an `id_employee` column, and none of them is cleaned or blocked when the employee row goes away. They include exactly the audit trails that make this hurt - `ps_module_history`, `ps_order_history`, `ps_log`, `ps_stock_mvt`, `ps_order_payment` and `ps_customer_message`, along with `ps_employee_session`, `ps_employee_shop`, `ps_message`, `ps_message_readed`, `ps_module_preference`, `ps_tab_module_preference`, `ps_supply_order_history`, `ps_supply_order_receipt_history` and `ps_warehouse`.

The precedent is in the same schema: `ps_customer` and `ps_address` both carry a `deleted` column and are hidden rather than removed. `ps_employee` is the outlier.

Worth knowing in the meantime, since it gets the outcome described here without any change: the employee grid already has an `active` toggle, so deactivating an employee instead of deleting them keeps the row and the id, and the account can no longer log in.
